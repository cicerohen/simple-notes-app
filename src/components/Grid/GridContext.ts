import React from 'react';
import { DEFAULT_PADDING } from '../../theme';

export interface GridContextInterface {
  columns: number;
  gutter: string;
}

const GridContext = React.createContext<GridContextInterface>({
  columns: 12,
  gutter: DEFAULT_PADDING
});

export default GridContext;
