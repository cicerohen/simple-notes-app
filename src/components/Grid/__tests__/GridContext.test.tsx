import React from 'react';
import { cleanup, render } from '@testing-library/react';
import GridContext from '../GridContext';

describe('<GridContext/> tests', () => {
  let component;
  beforeEach(() => {
    component = render(
      <GridContext.Consumer>
        {value => (
          <div>
            <p>columns:{value.columns}</p>
            <p>gutter:{value.gutter}</p>
          </div>
        )}
      </GridContext.Consumer>
    );
  });

  afterAll(cleanup);

  it('should have a correctly default value', () => {
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
