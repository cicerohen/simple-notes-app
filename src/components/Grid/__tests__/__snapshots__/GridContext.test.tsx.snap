// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContext/> tests should have a correctly default value 1`] = `
<div>
  <p>
    columns:
    12
  </p>
  <p>
    gutter:
    10px
  </p>
</div>
`;
